### PR TITLE
Mastermind method

### PR DESCRIPTION
**Pull request by adomas-s**

### CODE REVIEW

File: spec/mastermind_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require "mastermind"
+
+RSpec.describe Mastermind do
+  describe "#mastermind" do
+    let(:output) { StringIO.new }
+
+    it "won all guesses" do
+      guesses = ["RED", "GREEN", "BLUE", "YELLOW"]
+      input = String.new(guesses.join("\n"))
```
Comment (adomas-s) [inline spec/mastermind_spec.rb:9]: There's no need to do `String.new(...)`, a simple `guesses.join("\n")` is sufficient

File: lib/mastermind.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    current_attempt = 1
+    correct = 0
+    while correct < guess_attempts
+      guess = input.split("\n")
+
+      if guess.length != 4
+        raise ArgumentError, "Number of guess not completed, try again!!"
+      end
+
+      guess1 = guess[0]
```
Comment (adomas-s) [inline lib/mastermind.rb:22]: Try to extract the concept of what the number 4 means here, so you wouldn't have to enumerate each guess with a suffix of a number. Any chance it's `@guess_attempts`?

File: lib/mastermind.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
+class Mastermind
+  attr_reader :passcode, :input, :guess_attempts, :output
+  def initialize(passcode:, input: $stdin, output: $stdout, guess_attempts: 4)
+    @passcode = passcode
+    # @correct_answers = 0
+    @input = input
+    @guess_attempts = 5
```
Comment (adomas-s) [inline lib/mastermind.rb:7]: This method has a `guess_attempts` parameter, and it looks like it could be used here.

**Review by adomas-s — COMMENTED** [review]
You're on the right track 👍

File: spec/mastermind_spec.rb (hunk 1 above)
Comment (Fisola91) [reply spec/mastermind_spec.rb:9]: Noted

File: lib/mastermind.rb (hunk 2 above)
Comment (Fisola91) [reply lib/mastermind.rb:22]: Yes, chances is what I meant by guess_attempts

**Pull request merged**